examples/type-processing/README.md

```diff
@@ -0,0 +1,56 @@
+## Overview
+
+This sample demonstrates the supported types of the Ballerina MI module SDK. A Ballerina function with parameters and return type of `boolean`, `string`, `int`, `float`, `decimal`, `xml`, and `json` is supported by the MI SDK.
+This scenario illustrates a content-based routing use case where a JSON payload containing a `type` property is sent to an API. Based on the value of this `type` property, a corresponding Ballerina function is invoked.
+
+A json payload of following format is sent to the endpoint
+
+```json
+{
+    "type": "integer",
+    "val": 21
+}
+```
+
+For the type field it is possible to send `boolean`, `string`, `int`, `float`, `decimal`, `xml`, and `json`.
+
+This would output a result in the following format.
+
+```json
+{
+   "result": 42
+}
+```
+
+## Steps to Invoke the Sample
+
+Follow these steps to invoke the sample using the module:
+
+1. The `bal-type-processing` folder contains the Ballerina code for the module. Invoke the following command to generate the module:
+
+    ```bash
+    bal mi-module-gen <ballerina-project>
```

Comment:
`<ballerina-project>` should be bal-type-processing right? so lets give  something like `<ballerina-project, in this sample it is bal-type-processing>`